snuba/datasets/__init__.py

```diff
@@ -59,13 +64,35 @@ def column_expr(self, column_name, body):
         """
         Return an expression for the column name. Handle special column aliases
         that evaluate to something else.
-
         """
-        return column_name
+        return escape_col(column_name)
 
     def get_bulk_loader(self, source, dest_table):
         """
         Returns the instance of the bulk loader to populate the dataset from an
         external source when present.
         """
         raise NotImplementedError
+
+
+class TimeSeriesDataset(Dataset):
+    def __init__(self, *args, time_group_columns: Mapping[str, str], **kwargs):
+        super().__init__(*args, **kwargs)
+        # Convenience columns that evaluate to a bucketed time. The bucketing
+        # depends on the granularity parameter.
+        self.__time_group_columns = time_group_columns
```

Comment:
Might also make sense to assert that these columns actually exist in the schema.